### PR TITLE
push: add testing

### PR DESCRIPTION
**Pull request by twilly**
Push never had testing before. Now it does!

### CODE REVIEW

**Comment by twilly** [comment]
Pending landing of  #1560,  #1561, and #1562.

File: private/buf/cmd/buf/command/push/push_test.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+// Push pushes.
+func (m *mockPushService) Push(
+	_ context.Context,
+	req *connect_go.Request[registryv1alpha1.PushRequest],
+) (*connect_go.Response[registryv1alpha1.PushResponse], error) {
+	owner := req.Msg.Owner
+	cb, ok := m.callbacks[owner]
+	if ok {
+		cb(req.Msg)
+		m.called[owner] = struct{}{}
```
Comment (joshcarp) [inline private/buf/cmd/buf/command/push/push_test.go:70]: This fails `make racetest` here. 

Idk why it's not enabled on ci

File: private/buf/cmd/buf/command/push/push_test.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+					func(env map[string]string) map[string]string {
+						env["BUF_TOKEN"] = "invalid"
+						buftransport.SetDisableAPISubdomain(env)
+						injectConfig(t, appName, env)
+						return env
+					},
+				)(appName),
+				tarball(map[string][]byte{
+					"buf.yaml":  bufYAML(t, URL, owner, "repo"),
+					"foo.proto": nil,
```
Comment (joshcarp) [inline private/buf/cmd/buf/command/push/push_test.go:176]: what does it mean for a file to have `nil` contents in a tarball?

Comment (twilly) [reply private/buf/cmd/buf/command/push/push_test.go:176]: `nil` is the empty slice, hence the same as `[]byte{}`, and means an empty file. "foo.proto" and "bar.proto" are both empty files, which is fine for the tests written here. That may change if I get to testing when two proto files do not have identical content.

**Review by joshcarp — APPROVED** [review]
LGTM ✅

File: private/buf/cmd/buf/command/push/push_test.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+) {
+	t.Helper()
+	owner := strings.ReplaceAll(desc, " ", "_")
+	mock.respond(owner, resp)
+	mock.callback(owner, func(req *registryv1alpha1.PushRequest) {
+		assert.NotNil(t, req.Module, "missing module")
+	})
+	t.Run(desc, func(t *testing.T) {
+		t.Parallel()
+		appName := "test"
```
Comment (unmultimedio) [inline private/buf/cmd/buf/command/push/push_test.go:171]: This can be const

**Pull request merged**